### ISSUE TITLE
npm 3.1.1: npm install --no-shrinkwrap throws

### ISSUE THREAD

**Issue by mantoni** [npm/npm#8896]
Running `npm install --no-shrinkwrap` fails with this:

```
[develop]$ cat npm-debug.log
0 info it worked if it ends with ok
1 verbose cli [ 'node', '/usr/bin/npm', 'install', '--no-shrinkwrap' ]
2 info using npm@3.1.1
3 info using node@v0.10.38
4 silly loadCurrentTree Starting
5 silly install loadCurrentTree
6 silly install readLocalPackageData
7 verbose type undefined_method
8 verbose stack TypeError: Object #<Object> has no method 'loadArgMetadata'
8 verbose stack     at /usr/lib/node_modules/npm/lib/install.js:585:21
8 verbose stack     at /usr/lib/node_modules/npm/node_modules/graceful-fs/graceful-fs.js:76:16
8 verbose stack     at fs.js:272:14
8 verbose stack     at /usr/lib/node_modules/npm/node_modules/graceful-fs/graceful-fs.js:42:10
8 verbose stack     at Object.oncomplete (fs.js:108:15)
9 verbose cwd /path/to/my/project
10 error Linux 2.6.18-308.el5
11 error argv "node" "/usr/bin/npm" "install" "--no-shrinkwrap"
12 error node v0.10.38
13 error npm  v3.1.1
14 error Object #<Object> has no method 'loadArgMetadata'
15 error If you need help, you may report this error at:
15 error     <https://github.com/npm/npm/issues>
16 verbose exit [ 1, true ]
```

**Comment by iarna** [comment]
This will be fixed by #8922

**Comment by mantoni** [comment]
Works with `3.1.2` :+1:
Thanks!
